src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -48,28 +52,47 @@ public class ConsoleRunner
 
         public ConsoleRunner(ITestEngine engine, ConsoleOptions options, ExtendedTextWriter writer)
         {
-            _engine = engine;
-            _options = options;
-            _outWriter = writer;
-
-            _workDirectory = options.WorkDirectory ?? Directory.GetCurrentDirectory();
-
-            if (!Directory.Exists(_workDirectory))
-                Directory.CreateDirectory(_workDirectory);
+            Guard.ArgumentNotNull(_engine = engine, nameof(engine));
+            Guard.ArgumentNotNull(_options = options, nameof(options));
+            Guard.ArgumentNotNull(_outWriter = writer, nameof(writer));
 
+            // NOTE: Accessing Services triggerss the engine to initialize all services
             _resultService = _engine.Services.GetService<IResultService>();
+            Guard.OperationValid(_resultService != null, "Internal Error: ResultService was not found");
+
             _filterService = _engine.Services.GetService<ITestFilterService>();
+            Guard.OperationValid(_filterService != null, "Internal Error: TestFilterService was not found");
+
             _extensionService = _engine.Services.GetService<IExtensionService>();
+            Guard.OperationValid(_extensionService != null, "Internal Error: ExtensionService was not found");
 
-            // TODO: Exit with error if any of the services are not found
+            var extensionPath = Environment.GetEnvironmentVariable(NUNIT_EXTENSION_DIRECTORIES);
+            if (!string.IsNullOrEmpty(extensionPath))
+                foreach (string extensionDirectory in extensionPath.Split(new[] { Path.PathSeparator }, StringSplitOptions.RemoveEmptyEntries))
+                    _extensionService.FindExtensionAssemblies(extensionDirectory);
+
+            foreach (string extensionDirectory in _options.ExtensionDirectories)
+               _extensionService.FindExtensionAssemblies(extensionDirectory);
+
+            // Trigger lazy loading of extensions
+            var dummy = _extensionService.Extensions;
```

Comment:
Is this really needed? The service ensures these are loaded when needed.
If you want to keep it, I much prefer: `_extensionService.EnsureExtensionsAreLoaded()` over this assignment.

Reply:
Left over from earlier testing, before `EnsureExtensionsAreLoaded` existed.